### PR TITLE
LoadIsawDetCal parameter change

### PR DESCRIPTION
**Pull request by peterfpeterson**
Convert from having multiple `Filename` properties to a single property that accepts multiple filenames.

**To test:**

Since there was a modest modernization effort in the changes as well, seeing the parameter change by itself is hiding in the first commit.

Test by loading `.detcal` files onto instruments and seeing that it had an effect. There are some `.detcal` files in the system tests (often with the extension `.DetCal`).

*There is no associated issue.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by gvardany — CHANGES_REQUESTED** [review]
Could you add a unit test for multiple files case?

**Comment by VickieLynch** [comment]
Works with TOPAZ single file and SNAP with 2 files.  I shifted the first detector in each file and the correct detectors were moved for SNAP.

File: Framework/DataHandling/src/LoadIsawDetCal.cpp [hunk 1]
```diff
@@ -60,13 +62,50 @@ void LoadIsawDetCal::init() {
...
+    result["Filename"] = "Must supply .detcal file";
+  } else if (filenames.size() == 2) {
+    Workspace_const_sptr wksp = getProperty("InputWorkspace");
+    const auto instname = getInstName(wksp);
+
+    if (instname.compare("SNAP") != 0) {
+      result["Filename"] = "Two files is only valid for SNAP";
+    }
+  } else if (filenames.size() > 2) {
+    result["Filename"] = "Must supply .detcal file";
```
Comment (gvardany) [inline Framework/DataHandling/src/LoadIsawDetCal.cpp]: This message seems copy pasted from above.

File: Framework/DataHandling/src/LoadIsawDetCal.cpp [hunk 2]
```diff
@@ -458,15 +489,31 @@ Instrument_sptr LoadIsawDetCal::getCheckInst(API::Workspace_sptr ws) {
...
   }
 
   return inst;
 }
 
+std::vector<std::string> LoadIsawDetCal::getFilenames() {
+  std::vector<std::string> filenamesFromPropertyUnraveld;
+  std::vector<std::vector<std::string>> filenamesFromProperty =
+      this->getProperty("Filename");
+  for (auto outer : filenamesFromProperty) {
```
Comment (gvardany) [inline Framework/DataHandling/src/LoadIsawDetCal.cpp]: const auto & ?

**Review by gvardany — COMMENTED** [review]
Tests are a lot better now 👍 , see few more minor points inline.

**Review by gvardany — APPROVED** [review]

**Comment by gvardany** [comment]
Happy to :shipit: once tests pass

**Pull request merged**